### PR TITLE
Updated tests (Travis and Mocha) and dependencies, rebuilt dist

### PR DESCRIPTION
**Pull request by cwonrails**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

* Updated all dependencies, with the following ramifications:
  * Updating `gulp-autoprefixer` (which itself updated `autoprefixer` to v7) leads to a new prefix being added in `dist/css/main.css`.
  * Updating `mocha` to v4 required updating the `npm test` script (the `--compilers` flag was deprecated between v3 and v4). 
  * The `babel`-related flag in `npm test` was also updated to match current syntax.
* Several updates to: `.travis.yml`:
  * Tests now run on Ubuntu 14.04 (`trusty`) instead of 12.04 (`precise`, the previous default.)
  * Node 8 and 9 have been added and Node 7 has been removed as it has now reached EoL.
* Trimmed trailing whitespace in two lines of `CHANGELOG.md`
* Changes to `dist`:
  * In addition to the new css prefix, running a rebuild added previously-committed docs sections not reflected in `dist` (i.e. `humans.txt`) and added the `shrink-to-fit=no` meta tag to `dist/index.html`.

### CODE REVIEW

**Comment by roblarsen** [comment]
This looks great at first pass. I'll take a deeper look on Friday and pull this in. Thanks!

File: dist/doc/usage.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ serve as the foundation for whatever you're interested in building.
 
 Even the basic use-case of a simple static site can be enhanced by manipulating
 the code through an automated build process. Moving up in complexity HTML5
-Boilerplate can be integrated with whatever front-end framework, CMS or
+Boilerplate can be be integrated with whatever front-end framework, CMS or
```
Comment (sumityadav) [inline dist/doc/usage.md]: Duplicate? `be `

Comment (cwonrails) [reply dist/doc/usage.md]: Thanks for picking that up, will fix now - I noticed this was addressed manually in https://github.com/h5bp/html5-boilerplate/commit/37871286b4cb73d828eddf87c3f7f0d2e8b30a68 but that was in `dist/doc/usage.md` directly as opposed to during the build process - will track down the culprit right now.

Comment (cwonrails) [reply dist/doc/usage.md]: Fixed the error in `src/doc/usage.md` and rebuilt, thus fixing in `dist/doc/usage.md` as well.

**Pull request merged**